app/services/importer_service.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class ImporterService
+  def initialize(data_set_id, fields: {})
+    @data_set_id = data_set_id
+    @fields = fields
+    @per_page = 500
+    @total_records = fetch_total_records_count.to_i
+  end
+
+  def import!
+    offset = 0
+    page = 0
+    while offset < @total_records
+      ActiveRecord::Base.transaction do
+        offset = page * @per_page
+        fetch_records(@per_page, offset).each do |record|
+          save_org(record)
+        end
+        page += 1
+      end
+    end
+  end
+
+  private
+
+  def save_org(record)
+    puts record[@fields[:name]]
+    org = Organisation.find_or_create_by!(name: record[@fields[:name]])
```

Comment:
What happens if we don't set `@fields[:name]`? I think it will break - and it's pretty hidden for it to break here! 
Not sure about your approach with `@fields` here - does it even need to be this generic? Could we instead do:

```
def initialize(data_set_id:, name_field: 'Org_Name', email_field: 'Email', address_field: 'Add1_Line1', contact_number_field: 'Telephone')
```

Then if we need to overwrite them we can - but we've got enough defaults to make it work without having to dig through the internals of this service.

Reply:
now set in init()

Reply:
now set in the init()